modules/33acrossBidAdapter.js

```diff
@@ -152,6 +156,42 @@ function _createServerRequest(bidRequest, gdprConsent = {}, uspConsent, pageUrl)
   }
 }
 
+function _getSchainInfo(schain) {
+  if(_isValidSchain(schain)) {
+    return {
+      ext: { schain }
+    }
+  } else {
+    utils.logWarn('[33Across Adapter] Invalid schain info passed');
+    return null;
+  }
+}
+
+// We validate schain as follows:
+//  - Complete status must be 0|1
+//  - Nodes must be defined and non-empty array
+//  - Each node must contain all the required fields
+// The entire schain schain object is marked as invalid i.e. return false if any 
+// of the above conditions fail.
+function _isValidSchain(schain) {
+  //Check of nodes are valid
+  if(!schain.nodes || !schain.nodes.length > 0) {
```

Comment:
maybe change `!schain.nodes.length > 0` to `!(schain.nodes.length > 0)` or `schain.nodes.length < 1` for easy readability

Reply:
why is `schain.nodes.length < 1` more readable than `!schain.nodes.length > 0`?

Reply:
because `!schain.nodes.length > 0` will do this `(!schain.nodes.length) > 0`

Reply:
Yikes yes you are correct. Thank you.

Reply:
That's a bug then not just readability issue? Hmm how did my valid schain unit test pass then? Because `!1 > 0` evaluates to `false`?

Reply:
Actually nvm the above comment. It's a bug with no side-effects in this particular case because instead of evaluating `(1 > 0)` we are evaluating `(true > 0)` which is still incorrect logic even if it happens to work for us this time.

Reply:
because it will covert the boolean to number and then compare with 0, eg. `number(boolean(length))` so if not empty array, it will return 0, and empty array will return 1, which still satisfied the `x>0` logic

Reply:
Haha I replied first. But thanks for spotting this. Fixed by adding paren.

Reply:
lol, okok, you win...btw, are you keep refreshing the page??

Reply:
my page does not auto update the comment

Reply:
Ah no I did not refresh page. But I also notice that only a page refresh updates comments. Not like gitlab